api/include/opentelemetry/trace/global_provider.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <atomic>
+
+#include "opentelemetry/nostd/unique_ptr.h"
+#include "opentelemetry/trace/noop.h"
+#include "opentelemetry/trace/tracer_provider.h"
+
+namespace opentelemetry
+{
+namespace trace
+{
+
+class DefaultTracerProvider final : public opentelemetry::trace::TracerProvider
+{
+public:
+  opentelemetry::trace::NoopTracer *const GetTracer(
+      nostd::string_view library_name,
+      nostd::string_view library_version  = "",
+      const TracerOptions &tracer_options = {}) override
+  {
+    return tracer_.get();
+  }
+
+private:
+  DefaultTracerProvider()
+      : tracer_{nostd::unique_ptr<opentelemetry::trace::NoopTracer>(
+            new opentelemetry::trace::NoopTracer)}
+  {}
+  nostd::unique_ptr<opentelemetry::trace::NoopTracer> tracer_;
+
+  friend class Provider;
+};
+
+class Provider
```

Comment:
I think we should rename `global_provider.h` to `provider.h` to match the class name.